### PR TITLE
Switch to celluloid-based eventsource implementation

### PR DESCRIPTION
**Pull request by jkodumal**
This eliminates our dependency on EventMachine and replaces it with a Celluloid-based library for EventSource

### CODE REVIEW

File: ldclient-rb.gemspec [hunk 1]
```diff
@@ -30,5 +30,6 @@ Gem::Specification.new do |spec|
   spec.add_runtime_dependency "net-http-persistent", "~> 2.9"
   spec.add_runtime_dependency "concurrent-ruby", "~> 1.0.0"
   spec.add_runtime_dependency "hashdiff", "~> 0.2"
-  spec.add_runtime_dependency "ld-em-eventsource", "~> 0.2"
+  #spec.add_runtime_dependency "ld-em-eventsource", "~> 0.2"
```
Comment (drichelson) [inline ldclient-rb.gemspec]: Should this be removed instead of commented?

**Comment by drichelson** [comment]
Seems reasonable- Has Jeremy tried out this branch?

**Comment by jhubert** [comment]
Sorry guys. Been a crazy few weeks. Trying out this branch now.

**Comment by jhubert** [comment]
I haven't seen any issues in two days. Since there was no clearly repeatable test case I can't say for sure that the issue has been fixed. I'm going to switch back to try and recreate it with the EM version.

**Comment by drichelson** [comment]
Thanks for trying it out! we'll merge and release a new version in the coming days.

**Pull request merged**